src/front/glsl/parser/declarations.rs

```diff
@@ -199,8 +199,22 @@ impl<'source> ParsingContext<'source> {
             // TODO: Should we try to make constants here?
             // This is mostly a hack because we don't yet support adding
             // bodies to entry points for variable initialization
-            let maybe_constant =
-                init.and_then(|(root, meta)| parser.solve_constant(ctx.ctx, root, meta).ok());
+
+            // Initializers for external declarations must be constants,
+            // so if `solve_constant` failed, that's an error the user should see.
+            // But if it's an initializer for a local variable, then failing to
+            // produce a constant just means we emit the expression as code;
+            // that is not an error.
+            let maybe_constant = if let Some((root, meta)) = init {
+                let result = parser.solve_constant(ctx.ctx, root, meta);
+                match result {
+                    Ok(res) => Some(res),
+                    Err(err) if ctx.external => return Err(err),
```

Comment:
This will disallow shaders like:
```rust
#version 460

layout (location = 0) in vec4 read;
layout (location = 0) out vec4 fragColor;
vec4 a = read;

void main()
{
	fragColor = vec4(0.4, 0.4, 0.8, 1.0);
}
```
Which are valid glsl, you need to also check if the declaration is constant

Reply:
Oh, I didn't know that. Thanks for pointing it out.

Reply:
@JCapucho Thank you very much for the review! Just to clarify one thing to see if I'm following you, the code should assign `None` to `maybe_constant` instead of returning an error in that case?

In other words, is it only supposed to check if the expression of `root` is `Expression::Load{ .. }`?